examples/gateway-car/main.go

```diff
@@ -0,0 +1,72 @@
+package main
+
+import (
+	"flag"
+	"io"
+	"log"
+	"net/http"
+	"os"
+	"strconv"
+
+	"github.com/ipfs/go-blockservice"
+	offline "github.com/ipfs/go-ipfs-exchange-offline"
+	"github.com/ipfs/go-libipfs/gateway"
+	carblockstore "github.com/ipld/go-car/v2/blockstore"
+)
+
+func main() {
+	carFilePtr := flag.String("c", "", "path to CAR file to back this gateway from")
+	portPtr := flag.Int("p", 8080, "port to run this gateway from")
+	flag.Parse()
+
+	blockService, f, err := newBlockServiceFromCAR(*carFilePtr)
+	if err != nil {
+		log.Fatal(err)
+	}
+	defer f.Close()
+
+	gateway, err := newBlocksGateway(blockService)
+	if err != nil {
+		log.Fatal(err)
+	}
+
+	handler := newHandler(gateway, *portPtr)
+
+	address := ":" + strconv.Itoa(*portPtr)
+	log.Printf("Listening on %s", address)
```

Comment:
nit: can we make it listen on localhost by defaut and print a clickable URL?

Currently, it's listening on all IPs but prints only port:
```
./gateway -c docs.car -p 8040
2023/02/02 03:30:50 Listening on :8040
```

:thought_balloon: A neat example would be to print not only the gateway listener, but also direct links to roots from CAR header (roots are optional, but `dag export` always has one):

```
./gateway -c docs.car -p 8040
2023/02/02 03:30:50 Listening on https://localhost:8040
2023/02/02 03:30:50 Hosting CAR root at https://localhost:8040/ipfs/QmR2rs1JDUf9UBpLu5ibsT18ACVqRUBLFnP5q8nqoPR6m8
```